lib/logstash/codecs/cloudwatch_logs.rb

```diff
@@ -25,11 +31,13 @@ def decompress(data)
   end
 
   def parse(json, &block)
-    base = json.reject { |k,_| k == "logEvents" }.freeze
-    events = json["logEvents"]
+    events = json.delete("logEvents")
+    json.freeze
 
     events.each do |event|
-      yield LogStash::Event.new(base.merge(event))
+      epochmillis = event.delete("timestamp").to_i
+      event[LogStash::Event::TIMESTAMP] = LogStash::Timestamp.at(epochmillis / 1000, (epochmillis % 1000) * 1000)
```

Comment:
I was planning on leaving this to the `date` filter's `UNIX_MS` match setting. It keeps this codec simpler, and its also likely that the timestamp will be derived from a field other than the CloudWatch Log event's timestamp.